src/Simplex/Chat/Migrations/M20240102_note_folders.hs

```diff
@@ -0,0 +1,42 @@
+{-# LANGUAGE QuasiQuotes #-}
+
+module Simplex.Chat.Migrations.M20240102_note_folders where
+
+import Database.SQLite.Simple (Query)
+import Database.SQLite.Simple.QQ (sql)
+
+m20240102_note_folders :: Query
+m20240102_note_folders =
+  [sql|
+    CREATE TABLE note_folders (
+      note_folder_id INTEGER PRIMARY KEY AUTOINCREMENT,
+      user_id INTEGER NOT NULL REFERENCES users ON DELETE CASCADE,
+      created_at TEXT NOT NULL,
+      updated_at TEXT NOT NULL,
+      chat_ts TEXT NOT NULL,
+      favorite INTEGER NOT NULL DEFAULT 0,
+      unread_chat INTEGER DEFAULT 0 NOT NULL
+    );
+
+    ALTER TABLE chat_items ADD COLUMN note_folder_id INTEGER DEFAULT NULL REFERENCES note_folders ON DELETE CASCADE;
+    ALTER TABLE files ADD COLUMN note_folder_id INTEGER DEFAULT NULL REFERENCES note_folders ON DELETE CASCADE;
```

Comment:
add indexes:
- chat_items(note_folder_id)
- files(note_folder_id)
- note_folders(user_id)? probably excessive but we don't limit number of users and it won't hurt
- index for notes chat pagination - probably it would be smth like chat_items(user_id, note_folder_id, chat_item_id) once queries are changed there, but check last/before/after queries plans
- queries for chat list pagination would probably reuse existing indexes but also better to check plans for a few example queries

Reply:
It is using `user_id` index, then "temorary b-tree for ordering". Looks legit.